lib/os/spsc_pbuf.c

```diff
@@ -150,10 +151,25 @@ int spsc_pbuf_alloc(struct spsc_pbuf *pb, uint16_t len, char **buf)
 	}
 
 	cache_inv(rd_idx_loc, sizeof(*rd_idx_loc), flags);
-	__sync_synchronize();
+	barrier_dmem_fence_full();
```

Comment:
sorry to picky about this but is this barrier supposed to wait for the cache op to be completed? Because in that case you need `barrier_dsync_fence_full()` instead.